bundles/org.openhab.binding.feed/src/main/java/org/openhab/binding/feed/internal/handler/FeedHandler.java

```diff
@@ -59,67 +58,138 @@
 
     private void refreshFeedState() {
-        SyndFeed feed = fetchFeedData(urlString);
+        SyndFeed feed = fetchFeedData(configuration.URL);
         boolean feedUpdated = updateFeedIfChanged(feed);
 
         if (feedUpdated) {
+            publishDynamicChannelsIfLinked();
+
             List<Channel> channels = getThing().getChannels();
             for (Channel channel : channels) {
                 publishChannelIfLinked(channel.getUID());
             }
         }
     }
 
+    private void publishDynamicChannelsIfLinked() {
+        int numberOfEntries = configuration.numberOfEntries;
+
+        if (currentFeedState == null) {
+            // This will happen if the binding could not download data from the server
+            logger.trace("Cannot update channels; no data has been downloaded from the server!");
+            return;
+        }
+
+        List<SyndEntry> entries = currentFeedState.getEntries().stream().limit(numberOfEntries)
+                .collect(Collectors.toList());
+
+        setUnusedDynamicChannelsToUndef(entries.size());
+
+        for (int i = 0; i < entries.size(); i++) {
+            SyndEntry entry = entries.get(i);
+
+            ChannelUID titleChannelUID = new ChannelUID(titleChannelGroupUID, String.valueOf(i));
+            setChannelToState(titleChannelUID, getTitleState(entry));
+
+            ChannelUID descriptionChannelUID = new ChannelUID(descriptionChannelGroupUID, String.valueOf(i));
+            setChannelToState(descriptionChannelUID, getDescriptionState(entry));
+
+            ChannelUID dateChannelUID = new ChannelUID(dateChannelGroupUID, String.valueOf(i));
+            setChannelToState(dateChannelUID, getDateState(entry));
+        }
+    }
+
+    private void setUnusedDynamicChannelsToUndef(int amountOfUsedChannels) {
+        getThing().getChannelsOfGroup(titleChannelGroupUID.getId()).stream().skip(amountOfUsedChannels)
+                .forEach(channel -> updateState(channel.getUID(), UnDefType.UNDEF));
+
+        getThing().getChannelsOfGroup(descriptionChannelGroupUID.getId()).stream().skip(amountOfUsedChannels)
+                .forEach(channel -> updateState(channel.getUID(), UnDefType.UNDEF));
+
+        getThing().getChannelsOfGroup(dateChannelGroupUID.getId()).stream().skip(amountOfUsedChannels)
+                .forEach(channel -> updateState(channel.getUID(), UnDefType.UNDEF));
+    }
+
+    private void createDynamicChannels(int numberOfChannels) {
+        // All groups have the same number of channels, so it doesn't matter which one we pick
+        int existingChannels = getThing().getChannelsOfGroup(titleChannelGroupUID.getId()).size();
+
+        ThingBuilder thingBuilder = editThing();
+
+        for (int i = existingChannels; i < numberOfChannels; i++) {
+            Channel titleChannel = ChannelBuilder
+                    .create(new ChannelUID(titleChannelGroupUID, String.valueOf(i)), "String")
+                    .withLabel("Title " + (i + 1)).build();
```

Comment:
Why create a channel with an id number that is 1 less than in the label?

Reply:
I thought that it would make sense to use a zero-based index when interacting programmatically, but to use a 1-based index for the UI

Reply:
It might not be to much of an issue as the user can set the label on probably will. It might just be confusing when working with the channels created in the ui and then using them somewhere else, in items, rules or manual configuration. The user than has to make double check to use the right channel.